monitor/uidmonitor/processor.go

```diff
@@ -0,0 +1,398 @@
+				zap.L().Error("Failed to clean cache while destroying process",
+					zap.String("contextID", contextID),
+					zap.Error(err),
+				)
+			}
+
+			s.netcls.DeleteCgroup(cgroupPath)
+
+		} else {
+			s.netcls.DeleteCgroup(cgroupPath)
+			if err != nil {
+				zap.L().Error("Did not Find Context", zap.String("PUID", puid))
+			}
+
+			return nil
+		}
+		//s.Unlock()
+
+	}
+	s.netcls.Deletebasepath(contextID)
+	// Send the event upstream
+	if err := s.puHandler.HandlePUEvent(publishedContextID, monitor.EventDestroy); err != nil {
+		zap.L().Warn("Failed to clean trireme ",
+			zap.String("contextID", contextID),
+			zap.Error(err),
+		)
+	}
+
+	return nil
+}
+
+// Pause handles a pause event
+func (s *UIDProcessor) Pause(eventInfo *rpcmonitor.EventInfo) error {
+
+	contextID, err := s.generateContextID(eventInfo)
+	if err != nil {
+		return fmt.Errorf("Couldn't generate a contextID: %s", err)
+	}
+
+	return s.puHandler.HandlePUEvent(contextID, monitor.EventPause)
+}
+
+// ReSync resyncs with all the existing services that were there before we start
+func (s *UIDProcessor) ReSync(e *rpcmonitor.EventInfo) error {
+
+	deleted := []string{}
+	reacquired := []string{}
+	marktoPID := map[string][]string{}
+	defer func() {
+		if len(deleted) > 0 {
+			zap.L().Info("Deleted dead contexts", zap.String("Context List", strings.Join(deleted, ",")))
+		}
+		if len(reacquired) > 0 {
+			zap.L().Info("Reacquired contexts", zap.String("Context List", strings.Join(reacquired, ",")))
+		}
+	}()
+
+	walker, err := s.contextStore.WalkStore()
+	if err != nil {
+		return fmt.Errorf("error in accessing context store")
+	}
+	cgroups := cgnetcls.GetCgroupList()
+	for _, cgroup := range cgroups {
+		pidlist, _ := cgnetcls.ListCgroupProcesses(cgroup)
+		if len(pidlist) == 0 {
+			s.netcls.DeleteCgroup(cgroup)
+			continue
+		}
+		markval := cgnetcls.GetAssignedMarkVal(cgroup)
+		if list, ok := marktoPID[markval]; !ok {
+			marktoPID[markval] = pidlist
+		} else {
+			marktoPID[markval] = append(list, pidlist...)
+		}
+	}
+	for k, v := range marktoPID {
+		zap.L().Error("Context ID", zap.String("Key", k), zap.String("PIDS", strings.Join(v, ",")))
+	}
+	for {
+		contextID := <-walker
+		if contextID == "" {
+			break
+		}
+
+		storedPU := &StoredContext{}
+
+		if err := s.contextStore.GetContextInfo("/"+contextID, &storedPU); err != nil {
+			continue
+		}
+		eventInfo := storedPU.EventInfo
+		mark := storedPU.MarkVal
+		if pids, ok := marktoPID[mark]; !ok {
+			//No pids with stored mark destroy the context record and go to next context
+			s.contextStore.RemoveContext("/" + contextID)
+			continue
+		} else {
+			for _, pid := range pids {
+				eventInfo.PID = pid
+				s.Start(eventInfo)
+			}
+		}
+
+	}
+
+	return nil
+}
+
+// generateContextID creates the contextID from the event information
+func (s *UIDProcessor) generateContextID(eventInfo *rpcmonitor.EventInfo) (string, error) {
+
+	contextID := eventInfo.PUID
+	if eventInfo.Cgroup != "" {
```

Comment:
do the opposite. try to keep the if block as small as possible. In general avoid `else` as much as possible (in 90% of the cases you don’t need else). Also exit functions as early as possible